### ISSUE TITLE
Unable to inspect all slash command arguments

### ISSUE THREAD

**Issue by GunnyWaffle** [replaceitem/carpet-discarpet#31]
This took a lot of digging, testing, and triple checking. So correct me if I am wrong, but I believe that replacing `command` with `command_name` in commit 2829267bca4ed59a1c1ddb2eb8b4ff39ec76add1 introduced a regression.

While it is true that `command_name` is identical to `command:0`, yielding the "root name" of the command.
e.g. `/cmd a` sees `command_name` as `cmd`
I am unable to inspect if `/cmd a` or `/cmd b` is used, since neither populate `arguments` with a `SUB_COMMAND` type `Slash command interaction option` value.

If I run the following code and use `/cmd a v:test`, then the following output is printed to my console:
code:
```
__on_start() -> (
  task( _() -> (
    for (global_server~'slash_commands',
      dc_delete(_);
    );
    dc_create_slash_command({
      'name' -> 'cmd',
      'description' -> 'test',
      'options' -> [
        {
          'type' -> 'SUB_COMMAND',
          'name' -> 'a',
          'description' -> 'test a',
          'options' -> [
            {
              'type' -> 'STRING',
              'name' -> 'v',
              'description' -> 'test option',
              'required' -> true
            }
          ]
        },
        {
          'type' -> 'SUB_COMMAND',
          'name' -> 'b',
          'description' -> 'test b',
          'options' -> [
            {
              'type' -> 'STRING',
              'name' -> 'v',
              'description' -> 'test option',
              'required' -> true
            }
          ]
        }
      ]
    }, global_server);
  ));
);

__on_discord_slash_command(interaction) -> (
  task(_(i) -> (
    logger(i~'command_name');
    for (i~'arguments',
      logger('argument');
      logger(_~'name');
      logger(_~'is_subcommand_or_group');
      logger(_~'value');
    );
    dc_respond_interaction(i, 'RESPOND_IMMEDIATELY', 'Inspecting data, don\'t mind me.');
  ), interaction);
);
```
output:
```
[04:23:44] [pool-24-thread-1/INFO]: cmd
[04:23:44] [pool-24-thread-1/INFO]: argument
[04:23:44] [pool-24-thread-1/INFO]: v
[04:23:44] [pool-24-thread-1/INFO]: false
[04:23:44] [pool-24-thread-1/INFO]: test
```
This does not seem correct. I'm unable to determine if a or b were used. It's as if the `arguments` list consists of only the last thing parsed, rather than all parsed options.
I would instead expect the following output:
```
cmd
argument
a
true
dc_slash_command_interaction_option
argument
v
false
test
```
Let me know if I missed something here. Thanks!

EDIT: Worth noting, that my prior request for `arguments_by_name` is likely incompatible with what I expect here. Should nested sub-options share names. In this case, might be best to move that map to the option value as well such that each index of `arguments` has its own mapping. Or I am completely confused here haha!

**Comment by replaceitem** [comment]
Alright i fixed the `arguments` not returning all options in 1c335f9cd2a3390d15448d5828b02f01ce749802.
I'm not sure what to do about the fact that `arguments_by_name` can't work with duplicate slash command option names, but i think that's fine, can still use `arguments`

**Comment by replaceitem** [comment]
Released v1.3.2 now
